### PR TITLE
Kbv 34 implement getquestions endpoint

### PR DESCRIPTION
**Pull request by techaman-m**
Can you please approve the PR

### CODE REVIEW

**Comment by techaman-m** [comment]
**KBV End point:**  /question-request
**Example request payload:**

`{
    "firstName": " ",
    "surname": " ",
    "title": " ",
    "dateOfBirth": "",
    "addresses": [
        {
            "houseNameNumber": " ",
            "postcode": "",
            "townCity": " ",
            "street": " ",
            "addressType": "CURRENT"
        }
    ]
}`

**Response payload:**

`{
    "control": {
        "urn": "6ab0f316-2a0b-48e8-8103-c9aeefac1ffb",
        "authRefNo": "7ATLWBK6DH",
        "dateTime": null,
        "testDatabase": "A",
        "clientAccountNo": "J8193",
        "clientBranchNo": null,
        "operatorID": "GDSCABINETUIIQ01U",
        "parameters": {
            "oneShotAuthentication": "N",
            "storeCaseData": "P"
        }
    },
    "questions": {
        "question": [
            {
                "questionID": "Q00058",
                "text": "Which figure represents a credit limit on one of your active credit cards?",
                "tooltip": "Does not including Charge Cards.",
                "answerFormat": {
                    "identifier": "A00004",
                    "fieldType": "G ",
                    "answerList": [
                        "UP TO £1,000",
                        "OVER £1,000 UP TO £2,000",
                        "OVER £2,000 UP TO £3,000",
                        "OVER £3,000 UP TO £4,000",
                        "NONE OF THE ABOVE / DOES NOT APPLY"
                    ]
                },
                "answerHeldFlag": null
            },
            {
                "questionID": "Q00051",
                "text": "In which month and year did you open one of your current accounts?",
                "tooltip": "An account with a bank or similar organisation enabling payments to be made or received. Often comes with a debit card, chequebook and overdraft facility.",
                "answerFormat": {
                    "identifier": "A00004",
                    "fieldType": "G ",
                    "answerList": [
                        "03/1986",
                        "01/1986",
                        "07/1985",
                        "12/1988",
                        "NONE OF THE ABOVE / DOES NOT APPLY"
                    ]
                },
                "answerHeldFlag": null
            }
        ],
        "skipsRemaining": null,
        "skipWarning": null
    },
    "results": {
        "outcome": "Authentication Questions returned",
        "authenticationResult": null,
        "questions": null,
        "alerts": null,
        "nextTransId": {
            "string": [
                "RTQ"
            ]
        },
        "caseFoundFlag": null,
        "confirmationCode": null
    },
    "error": null
}`

File: src/main/java/uk/gov/di/cri/experian/kbv/api/gateway/KBVGateway.java [hunk 1]
```diff
@@ -1,22 +1,158 @@
...
+        System.out.println(saaResponse2);
+        QuestionsResponse result =
+                saaRequestMapper.mapSAAResponse2ToQuestionsResponse(saaResponse2);
+
+        return result;
+    }
+
+    private SAAResponse2 getSaaResponse2(
+            IdentityIQWebServiceSoap soapEndpoint, SAARequestDto saaRequestDto) {
+        SAARequest saaRequest = createRequest(saaRequestDto);
```
Comment (jkunle) [inline src/main/java/uk/gov/di/cri/experian/kbv/api/gateway/KBVGateway.java]: I think all this should be in the mapper and not in the gateway, and maybe call `SAARequestDto` as `QuestionRequest` and QuestionRequest should be moved to the Domain.

Comment (jkunle) [reply src/main/java/uk/gov/di/cri/experian/kbv/api/gateway/KBVGateway.java]: So `mapPersonIdentity` should return `SAARequest`

Comment (jkunle) [reply src/main/java/uk/gov/di/cri/experian/kbv/api/gateway/KBVGateway.java]: It doesn't even seem we need the `SAARequestDto/QuestionRequest` at all we could just have the `SAARequest mapPersonIdentity(createRequest(personIdentity))`

File: src/main/java/uk/gov/di/cri/experian/kbv/api/gateway/KBVGateway.java [hunk 2]
```diff
@@ -1,22 +1,158 @@
...
+            throws IOException, InterruptedException {
 
-        // TODO: make the call to the experian SAA endpoint
+        SAARequestDto apiRequest = saaRequestMapper.mapPersonIdentity(personIdentity);
 
-        return null;
+        IdentityIQWebServiceSoap identityIQWebServiceSoap =
+                kbvSoapWebServiceClient.getIdentityIQWebServiceSoapEndpoint();
+        SAAResponse2 saaResponse2 = getSaaResponse2(identityIQWebServiceSoap, apiRequest);
+        System.out.println(saaResponse2);
```
Comment (jkunle) [inline src/main/java/uk/gov/di/cri/experian/kbv/api/gateway/KBVGateway.java]: maybe we can remove this

File: src/main/java/uk/gov/di/cri/experian/kbv/api/domain/Question.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package uk.gov.di.cri.experian.kbv.api.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Question {
```
Comment (jkunle) [inline src/main/java/uk/gov/di/cri/experian/kbv/api/domain/Question.java]: Do we still need this? Along with Answers, FieldType I think this is superceeded by QuestionsResponse

**Review by jkunle — APPROVED** [review]

**Pull request merged**